### PR TITLE
behnaz/show synthetic before forex

### PR DESCRIPTION
**Pull request by behnaz-deriv**

### CODE REVIEW

File: src/javascript/_common/base/login.js [hunk 1]
```diff
@@ -8,11 +8,15 @@ const urlForCurrentDomain = require('../url').urlForCurrentDomain;
 const isLoginPages        = require('../utility').isLoginPages;
 const TrafficSource       = require('../../app/common/traffic_source');
 const getAppId            = require('../../config').getAppId;
+const Url                 = require('../../../javascript/_common/url');
```
Comment (negarn) [inline src/javascript/_common/base/login.js:11]: why are you importing it in such a weird way?

```suggestion
const Url                 = require('../url');
```
this should work, same as line 7, and you can replace line 7 with it and use `Url.urlForCurrentDomain` throughout the file instead

File: src/javascript/app/base/header.js [hunk 2]
```diff
@@ -63,10 +64,12 @@ const Header = (() => {
 
     const loginOnClick = (e) => {
         e.preventDefault();
+        Defaults.remove('market', 'underlying');
         Login.redirectToLogin();
     };
 
     const logoutOnClick = () => {
+        Defaults.remove('market', 'underlying');
```
Comment (negarn) [inline src/javascript/app/base/header.js:72]: can we move this to a separate function, something like `clearTradeParams` and add a comment to explain why we need to clear them?

File: src/javascript/app/pages/trade/common.js [hunk 3]
```diff
@@ -314,11 +314,12 @@ const commonTrading = (() => {
         if (tip) {
             const market = ActiveSymbols.getSymbols()[Defaults.get('underlying')].market;
             const map_to_section_id = {
+
```
Comment (negarn) [inline src/javascript/app/pages/trade/common.js:317]: empty line
```suggestion

```

File: src/javascript/app/base/header.js [hunk 4]
```diff
@@ -61,12 +62,19 @@ const Header = (() => {
         }
     };
 
+    // To clear trade parameters by removing market/underlying from url
+    const clearTradeParams = ()=> {
```
Comment (mustofa-binary) [inline src/javascript/app/base/header.js:66]:
```suggestion
    const clearTradeParams = () => {
```

File: src/javascript/app/pages/trade/markets.jsx [hunk 5]
```diff
@@ -177,7 +177,7 @@ class Markets extends React.Component {
         let curr_market = null;
         Object.entries(market_nodes).forEach(([key, node]) => {
 
-            if (node && node.offsetParent && node.offsetTop - 41 <= position) {
+            if (node && node.offsetParent && node.offsetTop - 100 <= position) {
```
Comment (mustofa-binary) [inline src/javascript/app/pages/trade/markets.jsx:180]: can you add comment to describe what is 100 here, or create a named variable instead 🙏

File: src/javascript/app/base/header.js (hunk 2 above)
Comment (behnaz-fs) [reply src/javascript/app/base/header.js]: Done

File: src/javascript/app/pages/trade/common.js (hunk 3 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/common.js]: Done

File: src/javascript/_common/base/login.js (hunk 1 above)
Comment (behnaz-fs) [reply src/javascript/_common/base/login.js]: Done

File: src/javascript/app/base/header.js (hunk 4 above)
Comment (behnaz-fs) [reply src/javascript/app/base/header.js:66]: done

Comment (behnaz-fs) [reply src/javascript/app/base/header.js:66]: done

File: src/javascript/app/pages/trade/markets.jsx (hunk 5 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:188]: comment added

**Comment by negarn** [comment]
@behnaz-fs please resolve the conflict

File: src/javascript/app/pages/trade/markets.jsx [hunk 6]
```diff
@@ -68,11 +68,19 @@ const List = ({
 class Markets extends React.Component {
     constructor (props) {
         super(props);
-        let market_symbol = Defaults.get('market');
+        let market_symbol,underlying_symbol;
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:71]: please check the syntax of how we declare variables. there should be a new line after the comma.

File: src/javascript/app/pages/trade/markets.jsx [hunk 7]
```diff
@@ -68,11 +68,19 @@ const List = ({
...
+        const login_id = localStorage.getItem('active_loginid');
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:72]: we usually get these values from Client, not from localstorage. please check the rest of the code on how loginid is usually retrieved.

File: src/javascript/app/pages/trade/markets.jsx [hunk 8]
```diff
@@ -68,11 +68,19 @@ const List = ({
...
+        if (localStorage.getItem(`${login_id}_selected_market`) && login_id) {
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:73]: this should be retrieved from Defaults, not from localstorage

Comment (negarn) [reply src/javascript/app/pages/trade/markets.jsx:73]: we would usually check Client.isLoggedIn instead, please check the codebase

File: src/javascript/app/pages/trade/markets.jsx [hunk 9]
```diff
@@ -68,11 +68,19 @@ const List = ({
...
+            market_symbol = localStorage.getItem(`${login_id}_selected_market`);
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:74]: you are retrieving this value twice, better store it in a variable instead

File: src/javascript/app/pages/trade/markets.jsx [hunk 10]
```diff
@@ -68,11 +68,19 @@ const List = ({
...
+        } else {
+            market_symbol = Defaults.get('market');
+        }
         this.markets = Symbols.markets();
 
         this.underlyings = Symbols.getAllSymbols() || {};
-        let underlying_symbol = Defaults.get('underlying');
+        if (localStorage.getItem(`${login_id}_selected_underlying`) && login_id) {
+            underlying_symbol = localStorage.getItem(`${login_id}_selected_underlying`);
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:82]: same here

File: src/javascript/app/pages/trade/markets.jsx [hunk 11]
```diff
@@ -197,9 +205,15 @@ class Markets extends React.Component {
     };
 
     onUnderlyingClick = (underlying_symbol, market_symbol) => {
-        Defaults.set('underlying', underlying_symbol);
-        Defaults.set('market', market_symbol);
 
+        const login_id = localStorage.getItem('active_loginid');
+        if (login_id) {
+            localStorage.setItem(`${login_id}_selected_underlying`,underlying_symbol);
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:211]: missing space after comma

File: src/javascript/app/pages/trade/markets.jsx (hunk 6 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:71]: 2 variables declared in two lines.

File: src/javascript/app/pages/trade/markets.jsx (hunk 9 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:74]: done

File: src/javascript/app/pages/trade/markets.jsx (hunk 10 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:82]: done

File: src/javascript/app/pages/trade/markets.jsx (hunk 11 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:211]: added

File: src/javascript/app/pages/trade/markets.jsx (hunk 7 above)
Comment (behnaz-fs) [reply src/javascript/app/pages/trade/markets.jsx:73]: checked with Client, but after logout selected market and underlying cleared, so after relogin cannot and retrieve these values by Client.

File: src/templates/app/_layout/head.jsx [hunk 12]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import Title from '../../_common/components/title.jsx';
+import Favicons from '../../_common/includes/favicons.jsx';
+import AntiClickjack from '../../_common/includes/anti_clickjack.jsx';
+import GoogleOptimizer from '../../_common/includes/google_optimizer.jsx';
```
Comment (negarn) [inline src/templates/app/_layout/head.jsx:5]: you're adding this file again, head has been moved to common folder. please remove this file and move your changes there.

File: src/javascript/app/pages/trade/markets.jsx [hunk 13]
```diff
@@ -68,11 +68,22 @@ const List = ({
 class Markets extends React.Component {
     constructor (props) {
         super(props);
-        let market_symbol = Defaults.get('market');
+        let market_symbol,
+            underlying_symbol;
+        const login_id = localStorage.getItem('active_loginid') ? localStorage.getItem('active_loginid') : 'logout';
```
Comment (negarn) [inline src/javascript/app/pages/trade/markets.jsx:73]: you can change this to
`localStorage.getItem('active_loginid') || 'logout'`

Comment (negarn) [reply src/javascript/app/pages/trade/markets.jsx:73]: can you also change `logout` to `logged_out`? I think it's clearer

**Pull request closed without merging**